pkg/skaffold/event/util.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package event
+
+import (
+	"strings"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/proto"
+)
+
+func initializeMetadata(p latest.Pipeline, kubeContext string) *proto.Metadata {
```

Comment:
WDYT of including a hash of the pipeline?  That would let us tell if the user re-ran or if they made a change.

Reply:
oh this is an interesting idea. you could even just translate that to a `isPipelineChange` boolean or something.